homeassistant/components/media_player/kodi.py

```diff
@@ -130,7 +138,11 @@ def async_setup_platform(hass, config, async_add_devices, discovery_info=None):
         password=config.get(CONF_PASSWORD),
         turn_off_action=config.get(CONF_TURN_OFF_ACTION), websocket=websocket)
 
+    hass.data[DATA_KODI].append(entity)
     async_add_devices([entity], update_before_add=True)
+    descriptions = yield from hass.loop.run_in_executor(
+        None, load_yaml_config_file, os.path.join(
+            os.path.dirname(__file__), 'services.yaml'))
 
```

Comment:
I think we want to add a guard clause here to check if the services are already registered. Something like
```python
if hass.services.has_service(DOMAIN, SERVICE_ADD_MEDIA):
    return
```